src/main/kotlin/cacophony/controlflow/FunctionHandler.kt

```diff
@@ -40,8 +41,82 @@ class FunctionHandlerImpl(
     override fun generateCall(
         arguments: List<CFGNode>,
         result: Register?,
+        respectStackAlignment: Boolean,
     ): CFGFragment {
-        TODO("Not yet implemented")
+        val registerArguments = arguments.zip(REGISTER_ARGUMENT_ORDER)
+        val stackArguments = arguments.drop(registerArguments.size).map { Pair(it, Register.VirtualRegister()) }
+
+        val nodes: MutableList<CFGNode> = mutableListOf()
+
+        if (respectStackAlignment) {
+            // we push two copies of RSP to the stack and either leave them both there,
+            // or remove one of them via RSP assignment
+            val oldRSP = Register.VirtualRegister()
+            nodes.add(CFGNode.Assignment(oldRSP, CFGNode.VariableUse(Register.FixedRegister(X64Register.RSP))))
+            for (i in 0..2) {
+                nodes.add(CFGNode.Push(CFGNode.VariableUse(oldRSP)))
+            }
```

Comment:
This pushes 3 times, because range expression is close-ended by default.

Reply:
Fixed. I unrolled the loop to make the code clearer.